### ISSUE TITLE
Expand all VS Code variables in robot.variables setting

### ISSUE THREAD

**Issue by DanielPenning** [robocorp/robotframework-lsp#701]
**Describe the bug**
Variables defined in the `robot.variables` setting can be used from within tests. 
If VS Code "core" variables are used inside the variable value, these variables are automatically expanded (replaced).
If variables from a VS Code extension are used, this is not the case.

This can also be seen as a feature request ;) I hope it is technically possible at all.

**To Reproduce**

Use the[ CMake extension](https://github.com/microsoft/vscode-cmake-tools) and define/select a build kit.
The variables `${buildKit}` and `${buildType}` are filled by VS Code.

```
    "cmake.buildDirectory": "${workspaceFolder}/target/build/${buildKit}-${buildType}",   // works
    "robot.variables": { 
          "build_dir": "${workspaceFolder}/build/${buildKit}-${buildType}",    // buildKit and buildType are not expanded
    }
```

**Expected behavior**
All variables are expanded for robot.variables values.

**Versions:**
 - OS: Windows 10
 - Robot Framework Version: 4.1.3
 - Robot Framework Language Server Version: v0.48.2
 - Client Version: VS Code 1.68.1

**Comment by fabioz** [comment]
I don't think this is possible at all... VSCode notably doesn't have an API for expanding variables, so, extensions are left scrambling to do the expansion themselves (see: https://github.com/microsoft/vscode/issues/46471).

So, in this case `cmake` can expand its own variables as it sees fit, but it's not possible for another extension to read `cmake` variables to expand those.
